daffodil-test/src/test/scala/org/apache/daffodil/section12/lengthKind/TestLengthKindExplicit.scala

```diff
@@ -98,14 +98,14 @@ class TestLengthKindExplicit {
     runner.runOneTest("denseBit_lengthKind_explicit")
   }
 
-  @Test def test_invalidLengthUnitsIntegerWarning_explicit(): Unit = {
-    runner.runOneTest("invalidLengthUnitsIntegerWarning_explicit")
+  @Test def test_lengthUnitsBitsForInteger_explicit(): Unit = {
+    runner.runOneTest("lengthUnitsBitsForInteger_explicit")
   }
-  @Test def test_invalidLengthUnitsIntegerError_explicit(): Unit = {
-    runner.runOneTest("invalidLengthUnitsIntegerError_explicit")
+  @Test def test_lengthUnitsBitsForInteger_explicit2(): Unit = {
+    runner.runOneTest("lengthUnitsBitsForInteger_explicit2")
   }
-  @Test def test_invalidLengthUnitsDecimalWarning_explicit(): Unit = {
-    runner.runOneTest("invalidLengthUnitsDecimalWarning_explicit")
+  @Test def lengthUnitsBitsForDecimal_explicit(): Unit = {
+    runner.runOneTest("lengthUnitsBitsForDecimal_explicit")
   }
 
```

Comment:
We might want some tests for xs:double, xs:float, and xs:date/time/dateTime with binaryCalendarRep set to binarySeconds and binaryMilliseconds.

For example, `binaryValue` in `ElementBaseGrammarMixin.scala` has logic for knownLength float/double, but I'm not sure it handles an expression for dfdl:length, unless that's handled somewhere else. Maybe we don't support expressions at all since those need to be 32/64 bits exactly?

Also, in that same val, the date/time/dateTime case errors that require lenghtUnits must be bytes for binarySeconds/Milliseconds, which this PR should fix, as long as the length is correct number of bits, we sholudn't care whether it was specified in bits or bytes.

Reply:
Added additional tests, and we don't support runtime valued length for binary float/double.